db/changes.go

```diff
@@ -524,8 +524,11 @@ func (db *Database) MultiChangesFeed(chans base.Set, options ChangesOptions) (<-
 		return nil, nil
 	}
 
-	if (options.Continuous || options.Wait) && options.Terminator == nil {
-		base.WarnfCtx(db.Ctx, "MultiChangesFeed: Terminator missing for Continuous/Wait mode")
+	if options.ChangesCtx == nil {
+		if options.Continuous || options.Wait {
+			base.WarnfCtx(db.Ctx, "MultiChangesFeed: Changes Context missing for Continuous/Wait mode")
+		}
+		options.ChangesCtx = context.Background()
```

Comment:
I'm wondering if this is a situation where we _want_ to find out about this (at dev time) by returning an error/panicing, rather than warning and carrying on with no way to kill the changes goroutines that are started up.

Looks like this is ancient code [if you step back far enough](https://github.com/couchbase/sync_gateway/commit/6d381ffea63ffdbf2efbfe0337ac68de95255334) and I've neither seen this warning anywhere before, nor do I think it's a risk for us to start to be stricter about this.

Reply:
The main point of adding this context is because lots of tests call the changes feed directly, bypassing the context initialisation. I will remove this changes context check, and change lots of tests to have the context added in the changes options.

Reply:
Understood. I think it makes sense to force the tests to pass in a context, otherwise we'd potentially never stop changes feeds as we have a non-cancellable background context in use.